src/lib/sql/sql_translator.cpp

```diff
@@ -578,6 +578,7 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_table_origin(const hsq
 
       } else if (Hyrise::get().storage_manager.has_view(hsql_table_ref.name)) {
         const auto view = Hyrise::get().storage_manager.get_view(hsql_table_ref.name);
+        _cacheable = false;
```

Comment:
1. to avoid a bug when caching (in the pqp cache) selected queries that use views: when you drop the view after the query and send the same query again, hyrise should throw an error, instead it executes the cached pqp.
2. https://github.com/hyrise/hyrise/pull/2031#issuecomment-605879687